extensions/logs/suite.go

```diff
@@ -0,0 +1,214 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var config Config
+
+type Config struct {
+	ArtifactsDir   string        `default:"logs" desc:"Directory for storing container logs" split_words:"true"`
+	ExcludeK8sNs   string        `default:"kube-system,local-path-storage" desc:"Comma-separated list of excluded kubernetes namespaces" split_words:"true"`
+	ContextTimeout time.Duration `default:"15s" desc:"Context timeout for kubernetes queries" split_words:"true"`
+	LogWorkerCount int           `default:"4" desc:"Number of log collector workers" split_words:"true"`
+}
+
+type Suite struct {
+	suite.Suite
+
+	kubeClient kubernetes.Interface
+
+	testStartTime time.Time
+	nsmContainers map[types.UID]bool
+	logQueue      chan logItem
+	waitGroup     sync.WaitGroup
+}
+
+type logItem struct {
+	namespace  string
+	pod        string
+	logDir     string
+	logOptions *corev1.PodLogOptions
+}
+
+func init() {
+	if err := envconfig.Usage("", &config); err != nil {
+		panic(err)
+	}
+
+	if err := envconfig.Process("", &config); err != nil {
+		panic(err)
+	}
+}
+
+func (s *Suite) SetupSuite() {
+	var err error
+
+	s.kubeClient, err = newKubeClient()
+	require.NoError(s.T(), err)
+
+	s.logQueue = make(chan logItem)
+	for i := 0; i < config.LogWorkerCount; i++ {
+		s.waitGroup.Add(1)
+		go func() {
+			defer s.waitGroup.Done()
+			for src := range s.logQueue {
+				s.saveLog(src)
+			}
+		}()
+	}
+}
+
+func (s *Suite) TearDownSuite() {
+	close(s.logQueue)
+	s.waitGroup.Wait()
+}
+
+func (s *Suite) SetupTest() {
+	s.testStartTime = time.Now()
+	s.nsmContainers = make(map[types.UID]bool)
+
+	ctx, cancel := context.WithTimeout(context.Background(), config.ContextTimeout)
+	defer cancel()
+
+	pods, err := s.kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+	require.NoError(s.T(), err)
+
+	for podIdx := range pods.Items {
+		pod := &pods.Items[podIdx]
+
+		if !isExcludedNamespace(pod.Namespace) {
+			s.nsmContainers[pod.UID] = true
+		}
+	}
+}
+
+func (s *Suite) AfterTest(suiteName, testName string) {
+	logDir := filepath.Join(config.ArtifactsDir, suiteName, testName)
+	require.NoError(s.T(), os.MkdirAll(logDir, os.ModePerm))
+
+	logOptions := corev1.PodLogOptions{
+		Timestamps: true,
+		SinceTime:  &metav1.Time{Time: s.testStartTime},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), config.ContextTimeout)
+	defer cancel()
+
+	pods, err := s.kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
```

Comment:
Interesting is `""` mean all namespaces? Please consider making this a constant `allNamespaces`